source/common/common/logger.h

```diff
@@ -416,17 +419,40 @@ template <Id id> class Loggable {
 
 #define ENVOY_LOG_EVERY_NTH(LEVEL, N, ...)                                                         \
   do {                                                                                             \
-    static auto* count = new std::atomic<uint64_t>();                                              \
-    if ((count->fetch_add(1) % N) == 0) {                                                          \
-      ENVOY_LOG(LEVEL, ##__VA_ARGS__);                                                             \
+    if (ENVOY_LOG_COMP_LEVEL(ENVOY_LOGGER(), LEVEL)) {                                             \
+      static auto* count = new std::atomic<int64_t>();                                             \
```

Comment:
Please switch these back to `uint64_t`. There is no possibility of overflow in 500+ years if we're bumping once a nanosecond..

Reply:
Done in 15819e6, except in the case where it's implied via the chrono count() call.